### ISSUE TITLE
response_body saving bytes instead of a decoded string

### ISSUE THREAD

**Issue by jamesvandyne** [easy-as-python/django-webmention#23]
I noticed a subtle bug in the [resolution.py::fetch_and_validate_source](https://github.com/easy-as-python/django-webmention/blob/9faab3dad9529d14399eb8dad016370adbd3054a/src/webmention/resolution.py#L19-L27).

While response.content is a bytes-string ( [docs](https://requests.readthedocs.io/en/master/user/quickstart/#binary-response-content)), it's never decoded before saving to the database. As a result it saves as a string into the db as follows: `"b'\n<!DOCTYPE html><html><head><meta charset="utf-8" />\n<...."`. 

Practically speaking this is fine for pure ascii data, but any unicode characters aren't be properly decoded before saving, resulting in double-escaped characters and must be corrected before you can display it properly to the user. e.g. `"Çelik"` will be saved as `"\xc3\x87elik"`

```
>>> b'\xc3\x87elik'  # raw bytes from response.content
b'\xc3\x87elik'
>>> str(b'\xc3\x87elik')  # raw string conversion gets double esacped
"b'\\xc3\\x87elik'" 
>>> b'\xc3\x87elik'.decode("utf-8")  # properly decoded
'Çelik'
```

Existing response_bodies can be fixed using [ftfy](https://ftfy.readthedocs.io/en/latest/) package by running the string through something like this `ftfy.fix_text(ftfy.fixes.decode_escapes(mention.response_body))`. 

Changing fetch_and_validate_source to return `response.text`, which is simply the decoded response.content would fix this issue.  Looking for the target in `str(response.content)` might also have a subtle bug if you used a utf-8 domain as well, so I believe it should validate using response.text as well.  

Refactored I think it should look something like this:

```python
def fetch_and_validate_source(source, target):
    response = requests.get(source)
    if response.status_code == 200:
        if target in response.text:
            return response.text
        else:
            raise TargetNotFoundError("Source URL did not contain target URL")
    else:
        raise SourceFetchError("Could not fetch source URL")
```

**Comment by daneah** [comment]
Thank you very much for the report and suggested solution @jamesvandyne—I will see if I can get a PR and release out for this this week.

**Comment by daneah** [comment]
@jamesvandyne this fix is released in [v2.0.2](https://pypi.org/project/django-webmention/2.0.2/)!

**Comment by jamesvandyne** [comment]
Thank you @daneah!
